tests/rpc/rpc_02-author_test.go

```diff
@@ -17,15 +17,133 @@
 package rpc
 
 import (
+	"bytes"
 	"fmt"
 	"os"
 	"testing"
 	"time"
 
+	"github.com/ChainSafe/gossamer/lib/common"
 	"github.com/ChainSafe/gossamer/tests/utils"
+	gsrpc "github.com/centrifuge/go-substrate-rpc-client/v2"
+	"github.com/centrifuge/go-substrate-rpc-client/v2/scale"
+	"github.com/centrifuge/go-substrate-rpc-client/v2/signature"
+	"github.com/centrifuge/go-substrate-rpc-client/v2/types"
 	"github.com/stretchr/testify/require"
 )
 
+func TestAuthorSubmitExtrinsic(t *testing.T) {
+	if utils.MODE != rpcSuite {
+		_, _ = fmt.Fprintln(os.Stdout, "Going to skip RPC suite tests")
+		//return
+	}
+
+	t.Log("starting gossamer...")
+
+	utils.CreateConfigBabeMaxThreshold()
+
+	nodes, err := utils.InitializeAndStartNodes(t, 1, utils.GenesisDefault, utils.ConfigDefault)
+	require.NoError(t, err)
+
+	defer func() {
+		t.Log("going to tear down gossamer...")
+		os.Remove(utils.ConfigDefault)
+		errList := utils.TearDown(t, nodes)
+		require.Len(t, errList, 0)
+	}()
+
+	time.Sleep(10 * time.Second) // wait for server to start
+
+	api, err := gsrpc.NewSubstrateAPI(fmt.Sprintf("http://localhost:%s", nodes[0].RPCPort))
+	require.NoError(t, err)
+
+	meta, err := api.RPC.State.GetMetadataLatest()
+	require.NoError(t, err)
+
+	// Create a call, transferring 12345 units to Bob
+	bob, err := types.NewAddressFromHexAccountID("0x90b5ab205c6974c9ea841be688864633dc9ca8a357843eeacf2314649965fe22")
+	require.NoError(t, err)
+
+	c, err := types.NewCall(meta, "Balances.transfer", bob, types.NewUCompactFromUInt(12345))
+	if err != nil {
+		panic(err)
+	}
+
+	// Create the extrinsic
+	ext := types.NewExtrinsic(c)
+
+	// TODO: the genesis hash is wrong.
+	// genesisHash, err := api.RPC.Chain.GetBlockHash(0)
+	// require.NoError(t, err)
+
+	rv, err := api.RPC.State.GetRuntimeVersionLatest()
+	require.NoError(t, err)
+
+	key, err := types.CreateStorageKey(meta, "System", "Account", signature.TestKeyringPairAlice.PublicKey, nil)
+	require.NoError(t, err)
+
+	var nonce uint32
+	ok, err := api.RPC.State.GetStorageLatest(key, &nonce)
+	require.NoError(t, err)
+	require.True(t, ok)
+
+	// this is the actual genesis hash for GenesisDefault
+	genesisHash := types.NewHash(common.MustHexToBytes("0x03170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c111314"))
```

Comment:
and remove this, then use the extrinsic from this test in the unit test, and it should work